holmes/plugins/toolsets/newrelic.py

```diff
@@ -0,0 +1,172 @@
+    def get_parameterized_one_liner(self, params) -> str:
+        return f"newrelic GetLogs(app='{params.get('app')}', since='{params.get('since')}')"
+
+
+class GetTraces(BaseNewRelicTool):
+    def __init__(self, toolset: "NewRelicToolset"):
+        super().__init__(
+            name="newrelic_get_traces",
+            description="Retrieve traces from New Relic",
+            parameters={
+                "duration": ToolParameter(
+                    description="Minimum trace duration in seconds",
+                    type="number",
+                    required=True,
+                ),
+                "trace_id": ToolParameter(
+                    description="Specific trace ID to fetch details (optional)",
+                    type="string",
+                    required=False,
+                ),
+            },
+            toolset=toolset,
+        )
+
+    def invoke(self, params: Any) -> str:
+        duration = params.get("duration")
+        trace_id = params.get("trace_id")
+
+        if trace_id:
+            query_string = f"SELECT * FROM Span WHERE trace.id = '{trace_id}' and duration.ms > {duration * 1000} and span.kind != 'internal'"
+        else:
+            query_string = f"SELECT * FROM Span WHERE duration.ms > {duration * 1000} and span.kind != 'internal'"
+
+        query = {
+            "query": f"""
+            {{
+                actor {{
+                    account(id: {self.toolset.nr_account_id}) {{
+                        nrql(query: \"{query_string}\") {{
+                            results
+                        }}
+                    }}
+                }}
+            }}
+            """
+        }
+
+        url = "https://api.newrelic.com/graphql"
+        headers = {
+            "Content-Type": "application/json",
+            "Api-Key": self.toolset.nr_api_key,
+        }
+
+        response = requests.post(url, headers=headers, json=query)
+        logging.info(f"Getting newrelic traces longer than {duration}s")
+        if response.status_code == 200:
+            data = response.json()
+            return json.dumps(data, indent=2)
+        else:
+            return f"Failed to fetch traces. Status code: {response.status_code}\n{response.text}"
+
+    def get_parameterized_one_liner(self, params) -> str:
+        if "trace_id" in params and params["trace_id"]:
+            return f"newrelic GetTraces(trace_id='{params.get('trace_id')}')"
+        return f"newrelic GetTraces(duration={params.get('duration')})"
+
+
+class NewrelicConfig(BaseModel):
+    nr_api_key: Optional[str] = None
+    nr_account_id: Optional[str] = None
+
+
+class NewRelicToolset(Toolset):
+    nr_api_key: Optional[str] = None
+    nr_account_id: Optional[str] = None
+
+    def __init__(self):
+        super().__init__(
+            name="newrelic",
+            description="Toolset for interacting with New Relic to fetch logs and traces",
+            docs_url="https://docs.newrelic.com/docs/apis/nerdgraph-api/",
```

Comment:
Similar comment to the above regarding docs, URL should be similar to `https://docs.robusta.dev/master/configuration/holmesgpt/toolsets/newrelic.html`